### PR TITLE
Feature/oss 113 tabs

### PR DESCRIPTION
**Pull request by shaikhspeare**
## Context

Tabs for managing and removing entities such as instances and workspaces.

## Changes

This PR introduces the following changes:
* Added tabs
* Revamped the UI

## Tasks

I've completed the following tasks:
- [x] Signed the Contributers License Agreement (CLA)
- [ ] Added/updated tests
- [ ] Added/updated docs

### CODE REVIEW

**Comment by cjoy** [comment]
Looks like the build, test and lint pipeline is failing. Is it broken in master?

File: ui/app/index.tsx [hunk 1]
```diff
@@ -1,18 +1,25 @@
-import React from "react";
-import { render } from "react-dom";
-import { debugContextDevtool } from "react-context-devtool";
-import "antd/dist/antd.css";
+import React from 'react'
```
Comment (cjoy) [inline ui/app/index.tsx:1]: Are you using eslint? I think I added some linting config, which we should probably use for consistency

File: ui/app/lib/entity-store/entity-actions.ts [hunk 2]
```diff
@@ -69,15 +70,21 @@ export const createActions = <T>(
   removeEntity: (entityId: string) => {
     const { [entityId]: _, ...rest } = state.entities;
     setState({
-      ...state,
-      entities: { ...rest },
+        type: "add",
```
Comment (cjoy) [inline ui/app/lib/entity-store/entity-actions.ts:73]: Is this supposed to be add or remove?

Comment (cjoy) [reply ui/app/lib/entity-store/entity-actions.ts:73]: Oh nvm, I think I get it now

File: ui/app/pages/instances/instances.tsx [hunk 3]
```diff
@@ -1,88 +1,105 @@
...
+        }
+        const updateInstance = (key: string, value: string, instance: Instance) => {
+            addEntity({
+                ...instance,
+                [key]: value,
+            })
+        }
+        return instances
+            .filter((instance): instance is Entity<Instance> => instance !== undefined && instance.key.includes(filter))
+            .map((instance) => {
```
Comment (cjoy) [inline ui/app/pages/instances/instances.tsx:47]: nit: we can use optional chaining
`instance?.key.includes(filter)`

**Review by cjoy — CHANGES_REQUESTED** [review]
Added a few comments - minor changes (should take less than 10mins to address)

File: ui/app/context/project.tsx [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import { createEntityContext } from '../lib/entity-store';
+
+export type Project = {
```
[the remark this answers is not in the document]
Comment (shaikhspeare) [reply ui/app/context/project.tsx:9]: will add in future PR

**Pull request merged**